src/app/clusters/push-av-stream-transport-server/PushAVStreamTransportCluster.h

```diff
@@ -45,25 +53,41 @@ class PushAvStreamTransportServer : public DefaultServerCluster
      * @note The caller must ensure the delegate lives throughout the instance's lifetime
      */
     PushAvStreamTransportServer(EndpointId aEndpointId, BitFlags<PushAvStreamTransport::Feature> aFeatures) :
-        DefaultServerCluster({ aEndpointId, PushAvStreamTransport::Id }), mLogic(aEndpointId, aFeatures)
+        DefaultServerCluster({ aEndpointId, PushAvStreamTransport::Id }), mFeatures(aFeatures),
+        mSupportedFormats{ PushAvStreamTransport::SupportedFormatStruct{ PushAvStreamTransport::ContainerFormatEnum::kCmaf,
+                                                                         PushAvStreamTransport::IngestMethodsEnum::kCMAFIngest } }
     {}
 
-    PushAvStreamTransportServerLogic & GetLogic() { return mLogic; }
-
     void SetDelegate(PushAvStreamTransportDelegate * delegate)
     {
-        mLogic.SetDelegate(delegate);
-        if (delegate != nullptr)
+        if (delegate == nullptr)
         {
-            delegate->SetPushAvStreamTransportServer(this);
+            ChipLogError(Zcl, "Push AV Stream Transport : Trying to set delegate to null");
+            return;
         }
+
+        mDelegate = delegate;
+        mDelegate->SetPushAvStreamTransportServer(this);
     }
 
-    void SetTLSClientManagementDelegate(TlsClientManagementDelegate * delegate) { mLogic.SetTLSClientManagementDelegate(delegate); }
+    void SetTLSClientManagementDelegate(TlsClientManagementDelegate * delegate)
+    {
+        mTLSClientManagementDelegate = delegate;
+        if (mTLSClientManagementDelegate == nullptr)
+        {
+            ChipLogError(Zcl, "Push AV Stream Transport : Trying to set TLS Client Management delegate to null");
+            return;
+        }
+    }
 
     void SetTlsCertificateManagementDelegate(TlsCertificateManagementDelegate * delegate)
     {
-        mLogic.SetTlsCertificateManagementDelegate(delegate);
+        mTlsCertificateManagementDelegate = delegate;
+        if (mTlsCertificateManagementDelegate == nullptr)
```

Comment:
same complaint here: the message says "trying" (which implies failure) but this does not actually fail.